### DISCUSSION TITLE
Entities with CollisionShape do not show up in debug-3d if they do not have a GlobalTransform

### DISCUSSION THREAD

**Discussion by nicopap** [jcornaz/heron#156] [Q&A] [answered]
When I spawn a CollisionShape as a child with the `debug-3d` feature on, the outline of the hitbox doesn't show up.

Reproduction:
```rust
fn spawn(mut commands: Commands) {
    let mut rigid_body_entity = commands.spawn();
    commands.spawn()
        .insert(RigidBody::Dynamic)
        .insert(Transform::from_translation(Vec3::new(-400.0, 200.0, 0.0)))
        .insert(GlobalTransform::default())
        .with_children(|children| {
            // A first physics shape
            children.spawn_bundle((
                CollisionShape::Cuboid {
                    half_extends: Vec3::new(15.0, 15.0, 0.0),
                    border_radius: None,
                },
                Transform::default(),
            ));
        });
}
```

The workaround is to add a `GlobalTransform` to the child entity. This reproduction example is copied from the `collision_shapes_in_child_entity.rs` example and I removed the `GlobalTransform` from the `spawn_bundle` to demonstrate it.

There is a workaround, but it would be welcome for things to work out of the box.

I know how to fix it, PR coming :)

**Comment by jcornaz** [comment, marked as answer]
Hi.

That's intended behavior, and in fact it is not only a problem of the debug renderer, the rigid-body should not work at all.

If there is no `GlobalTransform`, the body doesn't have a global position/rotation. It is *nowhere*.

It's even properly documented. Both, the `README.md` and root crate documentation say:

> Spawn any bundle of your choice. Only make sure there is a `GlobalTransform`
